### ISSUE TITLE
[no-unused-vars] ignores local tokens (type,interface)

### ISSUE THREAD

**Issue by Hotell** [typescript-eslint/typescript-eslint#1062]
**Repro**

```JSON
{
  "rules": {
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": "warn"
  }
}
```

```TS
// Expect Warning ✅
import { SomeType } from './types'

// Expect Warning ❌
interface ImNotUsed { }
// Expect Warning ❌
type NotUsedAsWell = unknown

type Empty = void

export const dummy = (): Empty => { console.log('dummy') }
```

**Expected Result**

local tokens `ImNotUsed`, `NotUsedAsWell` marked as unused

**Actual Result**

`ImNotUsed`, `NotUsedAsWell`  is ignored

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.2.0` |
| `@typescript-eslint/parser`        | `2.2.0` |
| `TypeScript`                       | `3.6.4` |
| `ESLint`                           | `6.3` |
| `node`                             | `10` |
| `npm`                              | `X.Y.Z` |
